optuna/samplers/_tpe/sampler.py

```diff
@@ -29,7 +31,6 @@
 from optuna.trial import TrialState
 
 
-_CONSTRAINTS_KEY = "tpe:constraints"
```

Comment:
It looks that there is a breaking change here. What do you think?

Reply:
Is this considered a breaking change?

Reply:
One alternative approach is to keep the original `_CONSTRAINTS_KEY`. This would make it impossible to switch sampler for one study.

Reply:
Sorry. This change looks acceptable since the new logic works as expected with storages that stores previous constraints keys.

Reply:
This feature has been introduced by https://github.com/optuna/optuna/pull/3506 for (only?) about 30 days; we haven't released this as a part of the stable release. So it sounds okay to introduce this backward compatibility for this time because the effect is supposed to be limited.

Reply:
Thank you for your comment!
That sounds reasonable. We may omit data migration for this change.

Reply:
How about `nsgaii:constraints` and `botorch:constraints`?

Reply:
We decided that this change is acceptable because both of constrained optimizations of NSGAII and BoTorch are currently experimental.